src/widgets/sparkline.rs

```diff
@@ -153,57 +175,127 @@ impl<'a> Widget for Sparkline<'a> {
+        let mut buffer = Buffer::filled(Rect::new(0, 0, width, height), &cell);
+        widget.render(buffer.area, &mut buffer);
         buffer
     }
 
     #[test]
     fn it_does_not_panic_if_max_is_zero() {
         let widget = Sparkline::default().data(&[0, 0, 0]);
-        let buffer = render(widget, 6);
-        assert_eq!(buffer, Buffer::with_lines(vec!["   xxx"]));
+        let buffer = render(widget, 6, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["   xxx"]));
     }
 
     #[test]
     fn it_does_not_panic_if_max_is_set_to_zero() {
         let widget = Sparkline::default().data(&[0, 1, 2]).max(0);
-        let buffer = render(widget, 6);
-        assert_eq!(buffer, Buffer::with_lines(vec!["   xxx"]));
+        let buffer = render(widget, 6, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["   xxx"]));
     }
 
     #[test]
-    fn it_draws() {
+    fn it_renders() {
         let widget = Sparkline::default().data(&[0, 1, 2, 3, 4, 5, 6, 7, 8]);
-        let buffer = render(widget, 12);
-        assert_eq!(buffer, Buffer::with_lines(vec![" ▁▂▃▄▅▆▇█xxx"]));
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec![" ▁▂▃▄▅▆▇█xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_max_more_than_data_max() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .max(16);
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec!["  ▁▁▂▂▃▃▄xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_max_less_than_data_max() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .max(4);
+        let buffer = render(widget, 12, 1);
+        assert_buffer_eq!(buffer, Buffer::with_lines(vec![" ▂▄▆█████xxx"]));
+    }
+
+    #[test]
+    fn it_renders_with_multi_line() {
+        let widget = Sparkline::default().data(&[0, 1, 2, 3, 4, 5, 6, 7, 8]);
+        let buffer = render(widget, 15, 3);
+        assert_buffer_eq!(
+            buffer,
+            Buffer::with_lines(vec![
+                "      ▂▅█xxxxxx",
+                "   ▁▄▇███xxxxxx",
+                " ▃▆██████xxxxxx",
+            ])
+        );
+    }
+
+    #[test]
+    fn it_renders_with_multi_line_and_baseline() {
+        let widget = Sparkline::default()
+            .data(&[0, 1, 2, 3, 4, 5, 6, 7, 8])
+            .show_baseline(true);
+        let buffer = render(widget, 15, 3);
+        assert_buffer_eq!(
+            buffer,
+            // this currently fails because the baseline logic doesn't clear
+            // the parts above the line
+            // "      ▂▅█xxxxxx",
+            // "   ▁▄▇███xxxxxx",
+            // " ▃▆██████▁▁▁▁▁▁",
+            Buffer::with_lines(vec![
+                "      ▂▅█      ",
+                "   ▁▄▇███      ",
+                " ▃▆██████▁▁▁▁▁▁",
+            ])
+        );
```

Comment:
Added a failing test for this - I'm curious what @cjbassi might suggest is the correct implementation of this?